### PR TITLE
Right button drag to rotate

### PR DESCRIPTION
**Pull request by Pessimistress**
- Right button drag to rotate (match Mapbox behavior)
- Disable drag rotate on map controls

### CODE REVIEW

File: src/utils/map-controls.js [hunk 1]
```diff
@@ -86,6 +86,11 @@ export default class MapControls {
       srcEvent.ctrlKey || srcEvent.shiftKey);
   }
 
+  isRightButtonPressed(event) {
+    const {srcEvent: {which, buttons}} = event;
+    return buttons === undefined ? (which === 3) : Boolean(buttons & 2);
```
Comment (abmai) [inline src/utils/map-controls.js]: Can we add a comment to indicate what the logic here is? i.e. what is `3` (assuming it's the right mouse code), and what `buttons & 2` does.

File: src/utils/map-controls.js [hunk 2]
```diff
@@ -86,6 +86,11 @@ export default class MapControls {
       srcEvent.ctrlKey || srcEvent.shiftKey);
   }
 
+  isRightButtonPressed(event) {
```
Comment (ibgreen) [inline src/utils/map-controls.js]: Seems like something mjolnir should offer?

File: src/utils/map-controls.js (hunk 1 above)
Comment (abmai) [reply src/utils/map-controls.js]: Actually, do we need to use `&` here?

Comment (Pessimistress) [reply src/utils/map-controls.js]: [buttons](https://developer.mozilla.org/en-US/docs/Web/API/MouseEvent/buttons) is a sum of the values of all active buttons:

File: src/utils/map-controls.js (hunk 2 above)
Comment (Pessimistress) [reply src/utils/map-controls.js]: Ideally. The event may be generated from one of several listeners so the format is inconsistent. We need to do an API audit on mjolnir.js.

File: src/utils/map-controls.js (hunk 1 above)
Comment (abmai) [reply src/utils/map-controls.js]: I see, thanks! It's definitely worth leaving a comment then, especially for the `&` and how it's an accurate way of detecting the right mouse press.

**Pull request merged**